clients/src/main/java/org/apache/kafka/clients/consumer/internals/DefaultBackgroundThread.java

```diff
@@ -145,19 +166,46 @@ void runOnce() {
             this.inflightEvent = Optional.empty();
         }
 
+        // TODO: Add a condition here, like shouldFindCoordinator in the future.  Since we don't always need to find
+        //  the coordinator.
+        if (coordinatorUnknown()) {
+            coordinatorManager.tryFindCoordinator().ifPresent(networkClientDelegate::add);
```

Comment:
If `CoordinatorManager` is backing off after a failure, we want to ensure that the network thread gets woken up after the backoff has expired. Perhaps instead of returning `UnsentRequest` directly, we can create a new object:

```java
class PollResult {
  int millisUntilNextEvent;
  List<UnsentRequest> requestsToSend;
}
```
Then perhaps we could structure each manager class in a similar way. Instead of `tryFindCoordinator`, we could have a generic `poll()`. For example:
```java
class CoordinatorManager {
  PollResult poll(long currentTimeMs);
}
```